fix: improve recipe parsing in build workflows

HastD opened the pull request:
Use `yq` instead of `sed` to extract the image name and base image data from recipe files. This ensures correct parsing of the YAML so the parsed values match what BlueBuild reads from the recipe file, even in the presence of trailing whitespace, quotation marks around the string, or other similar syntax elements allowed by YAML but not handled by the previous parsing method.

Also cleaned up trailing whitespace in recipe files.

RoyalOughtness commented on .github/workflows/build-one-recipe.yml:
```diff
@@ -68,9 +68,9 @@ jobs:
-          echo "IMAGE_NAME=$(sed -n 's/^name: //p' "./recipes/${RECIPE}")" >> "${GITHUB_ENV}"
-          echo "BASE_IMAGE=$(sed -n 's/^base-image: //p' "./recipes/${RECIPE}")" >> "${GITHUB_ENV}"
-          echo "BASE_IMAGE_VERSION=$(sed -n 's/^image-version: //p' "./recipes/${RECIPE}")" >> "${GITHUB_ENV}"
+          echo "IMAGE_NAME=$(grep -oP '^name:\s+\K.*\S(?=\s*$)' "./recipes/${RECIPE}")" >> "${GITHUB_ENV}"
```
can we change this back to the original approach? It's more readable and was working for years 😅 :

```
          echo "IMAGE_NAME=$(grep '^name:' "./recipes/${RECIPE}" | sed 's/^name: //')" >> "${GITHUB_ENV}"

```

HastD replied:
The original approach had the same bug fixed in this PR: it didn't strip trailing whitespace. (This PR also handles extra whitespace between the colon and the value.)

A better approach if there's a good tool for it would be to use a proper YAML-parsing tool, and then we can simply query the values by key name. (That would also handle some cases that this regex doesn't, like if the string is enclosed in quotes.) Thoughts?

RoyalOughtness replied:
@HastD I'm missing something because the provenance was functioning before with this approach, so the bug is new.

For yaml parsing, there's `yq` (apt install yq)

HastD replied:
Switched to using `yq`. I think this is both more readable and more robust now.

HastD committed:
fix: improve recipe parsing in build workflows

Use `yq` instead of `sed` to extract the image name and base image data
from recipe files. This ensures correct parsing of the YAML so the
parsed values match what BlueBuild reads from the recipe file, even in
the presence of trailing whitespace, quotation marks around the string,
or other similar syntax elements allowed by YAML but not handled by the
previous parsing method.

Also cleaned up trailing whitespace in recipe files.

RoyalOughtness committed:
Merge branch 'live' into provenance-parsing
